thaw_utils/src/signals/model.rs

```diff
@@ -1,15 +1,15 @@
 use leptos::{
-    Memo, ReadSignal, RwSignal, Signal, SignalGet, SignalGetUntracked, SignalSet, SignalUpdate,
-    SignalWith, SignalWithUntracked, WriteSignal,
+    Memo, ReadSignal, RwSignal, Signal, SignalGet, SignalGetUntracked, SignalSet, SignalSetter,
+    SignalUpdate, SignalWith, SignalWithUntracked, WriteSignal,
 };
 
 pub struct Model<T>
 where
     T: 'static,
 {
     read: Signal<T>,
-    write: WriteSignal<T>,
-    on_write: Option<WriteSignal<T>>,
+    write: SignalSetter<T>,
```

Comment:
I think it would be better to change it to an enumeration containing WriteSignal and SignalSetter, so that when it is WriteSignal type, you don't have to clone the value in the update function.

Reply:
Makes sense!

Reply:
Added an enum type